src/client/clientlauncher.cpp

```diff
@@ -449,6 +449,11 @@ bool ClientLauncher::launch_game(std::string &error_message,
 		server_name = menudata.servername;
 		server_description = menudata.serverdescription;
 
+		/* make sure that password will not stay somewhere in memory */
+		for (size_t i = 0; i < menudata.password.size(); i++)
+			menudata.password[i] = '*';
+		menudata.password.clear();
```

Comment:
compilers can optimize this out, since it's a dead-store
You want something like [`explicit_bzero`](https://man.freebsd.org/cgi/man.cgi?query=explicit_bzero) instead

Reply:
fixed